### PR TITLE
added animal index page and animal MVC

### PR DESCRIPTION
**Pull request by mcat115**
Added index to routes
Made animal model (validates name, body, and rating)
Made animal migration  (animal has rating (which will be aggregate rating), body (for description of animal), and name)
Added index to routes
Made view for index
Made react router
Made index component

### CODE REVIEW

File: app/controllers/animals_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class AnimalsController < ApplicationController
```
Comment (dovakeen118) [inline app/controllers/animals_controller.rb:1]: This is not an API controller? Take a look at my other comments below where you have both a React component for the index and an ERB template. From this controller it looks like you intend to use the ERB template, in which case you can delete the component, or vice versa. In either case, your view (template or component) should be set up to display the animals data

File: app/javascript/react/components/AnimalIndex.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import React from 'react'
+const AnimalIndex = (props) => {
+  return (
+    <h1>Animals</h1>
```
Comment (dovakeen118) [inline app/javascript/react/components/AnimalIndex.js:4]: No animals being displayed yet? For this feature of an animals index page, we wouldn't consider it complete if we aren't able to see that data that you set up on the backend!

File: db/migrate/20210421141639_create_animals.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class CreateAnimals < ActiveRecord::Migration[5.2]
+  def change
+    create_table :animals do |t|
+      t.string :name, null: false
+      t.text :body, null: false
+      t.float :rating, default: 0, null: false
```
Comment (dovakeen118) [inline db/migrate/20210421141639_create_animals.rb:6]: TIMESTAMPS!!

File: app/views/animals/index.html.erb [hunk 4]
```diff
@@ -0,0 +1 @@
+<h1>This will be replaced by React</h1>
```
Comment (dovakeen118) [inline app/views/animals/index.html.erb:1]: You have both an index ERB and an index React component - I would choose one or the other

**Review by dovakeen118 — CHANGES_REQUESTED** [review]
Looks like a little more to be added for this feature - also make sure your migrations have timestamps!

File: db/migrate/20210421141639_create_animals.rb (hunk 3 above)
Comment (mcat115) [reply db/migrate/20210421141639_create_animals.rb:6]: I added timestamps last minute but didn't commit... smh

**Comment by mcat115** [comment]
Since the last pull request, we have:
Deleted the erbs what were not necessary because they were being handled by react
Committed again with timestamps in the commit this time
Added ApiController
Added a fetch request and useEffect to animal index.js
Added a map function to return animal tiles for when we make those

**Pull request merged**